### PR TITLE
Add unit tests for pkg azure/auth

### LINKED ISSUE #86301: [Cloud Provider] Add unit tests for pkg azure/auth
**What happened**:

There are no unit tests yet for pkg [azure/auth](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/azure/auth), we should add unit tests.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by stcheng**
Signed-off-by: Shu0t1an Cheng <shuche@microsoft.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
/kind flake

**What this PR does / why we need it**:
This PR adds the unit tests for azure/auth package

**Which issue(s) this PR fixes**:

Fixes #86301

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by stcheng** [comment]
signed the CLA

**Review by feiskyer — COMMENTED** [review]
/area provider/azure
/priority important-soon
/ok-to-test

File: staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	}
+	env := &azure.PublicCloud
+
+	token, err := GetServicePrincipalToken(config, env)
+	assert.NoError(t, err)
+
+	oauthConfig, err := adal.NewOAuthConfigWithAPIVersion(env.ActiveDirectoryEndpoint, config.TenantID, nil)
+	assert.NoError(t, err)
+
+	spt, err := adal.NewServicePrincipalToken(*oauthConfig, config.AADClientID, config.AADClientSecret, env.ServiceManagementEndpoint)
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go]: check err here

File: staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		token, err := GetServicePrincipalToken(config, env)
+		assert.NoError(t, err)
+
+		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		assert.NoError(t, err)
+
+		spt, err := adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(msiEndpoint,
+			env.ServiceManagementEndpoint, config.UserAssignedIdentityID)
+		assert.NoError(t, err)
+
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go]: nit: remove empty line

File: staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	for _, config := range configs {
+		token, err := GetServicePrincipalToken(config, env)
+		assert.NoError(t, err)
+
+		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		assert.NoError(t, err)
+
+		spt, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, env.ServiceManagementEndpoint)
+		assert.NoError(t, err)
+
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go]: nit: remove empty line

File: staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	}
+
+	for _, c := range cases {
+		env, err := ParseAzureEnvironment(c.cloudName, c.resourceManagerEndpoint, c.identitySystem)
+		assert.NoError(t, err)
+		assert.Equal(t, env, c.expected)
+	}
+}
+
+func TestAzureStackOverridesNoAdfs(t *testing.T) {
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth_test.go]: s/TestAzureStackOverridesNoAdfs/TestAzureStackOverrides

**Comment by stcheng** [comment]
@feiskyer comments resolved

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by feiskyer** [comment]
Please run `./hack/update-bazel.sh` to fix the verification tests

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**